recordedit/form.controller.js

```diff
@@ -310,7 +310,18 @@
 
             params.displayMode = vm.editMode ? recordsetDisplayModes.foreignKeyPopupEdit : recordsetDisplayModes.foreignKeyPopupCreate;
 
-            params.reference = column.filteredRef(submissionRow, vm.recordEditModel.foreignKeyData[rowIndex]).contextualize.compactSelect;
+            var andFilters = [];
+            // loop through all columns that make up the key information for the association with the leaf table and create non-null filters
+            column.foreignKey.key.colset.columns.forEach(function (col) {
```

Comment:
I don't know why I assumed this PR only includes the related table changes and missed this. If we're doing this for foreign key popup, you should also change the `inputSwitch` implementation to do the same for foreign key popups.